### PR TITLE
A function to get the current master key info.

### PR DESCRIPTION
**Pull request by codeforall**
Commit adds tde_master_key_info() function that returns the information about the master key for the database.

select * from tde_master_key_info();

 master_key_name | key_provider_name | key_provider_id | master_key_internal_name | master_key_version |      key_createion_time

-----------------+-------------------+-----------------+--------------------------+--------------------+-------------------------------

 test-master-key | file-two          |               1 | test-master-key_1        |                  1 | 2024-03-25 18:07:13.131834+05

(1 row)

### CODE REVIEW

File: sql/pgtde_is_encrypted.sql [hunk 1]
```diff
@@ -21,6 +21,9 @@ SELECT amname FROM pg_class INNER JOIN pg_am ON pg_am.oid = pg_class.relam WHERE
 SELECT pgtde_is_encrypted('test_enc');
 SELECT pgtde_is_encrypted('test_norm');
 
+SELECT  key_provider_id, key_provider_name, master_key_name
+		from tde_master_key_info();
```
Comment (dutow) [inline sql/pgtde_is_encrypted.sql:25]: Inconsistent case: `FROM`

File: src/catalog/tde_master_key.c [hunk 2]
```diff
@@ -31,7 +32,7 @@
 
 #define DEFAULT_MASTER_KEY_VERSION      1
 
-typedef struct TdeMasterKeySharedState
+        typedef struct TdeMasterKeySharedState
```
Comment (dutow) [inline src/catalog/tde_master_key.c:35]: unnecessary formatting change?

File: src/catalog/tde_master_key.c [hunk 3]
```diff
@@ -674,3 +675,63 @@ pg_tde_rotate_key(PG_FUNCTION_ARGS)
...
+    Datum values[6];
+    bool isnull[6];
+    HeapTuple tuple;
+    Datum result;
+    TDEMasterKey *master_key;
+    TimestampTz ts;
+    GenericKeyring *keyring;
+
+        /* Build a tuple descriptor for our result type */
+        if (get_call_result_type(fcinfo, NULL, &tupdesc) != TYPEFUNC_COMPOSITE)
```
Comment (dutow) [inline src/catalog/tde_master_key.c:692]: Also inconsistent indentation in this function

File: sql/pgtde_is_encrypted.sql [hunk 4]
```diff
@@ -21,6 +21,9 @@ SELECT amname FROM pg_class INNER JOIN pg_am ON pg_am.oid = pg_class.relam WHERE
 SELECT pgtde_is_encrypted('test_enc');
 SELECT pgtde_is_encrypted('test_norm');
 
+SELECT  key_provider_id, key_provider_name, master_key_name
```
Comment (dutow) [inline sql/pgtde_is_encrypted.sql:24]: Can we also have a test with the extension enabled, but without anything configured? (no provider / master key added)

File: src/catalog/tde_master_key.c [hunk 5]
```diff
@@ -674,3 +675,63 @@ pg_tde_rotate_key(PG_FUNCTION_ARGS)
...
+            ereport(ERROR,
+                    (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                     errmsg("function returning record called in context that cannot accept type record")));
+
+    master_key = GetMasterKey();
```
Comment (dutow) [inline src/catalog/tde_master_key.c:697]: `master_key` can be NULL

**Review by dutow — COMMENTED** [review]
Mostly minor formatting related comments, but I think we also might have an issue with an unconfigured pg_tde, please add a test about that too

File: sql/pgtde_is_encrypted.sql (hunk 4 above)
Comment (codeforall) [reply sql/pgtde_is_encrypted.sql:26]: done
